packages/generator/tests/test_assets/example_renders/measurement_plugin_client/test_measurement_client.py

```diff
@@ -0,0 +1,398 @@
+        if self._stub is None:
+            with self._initialization_lock:
+                if self._grpc_channel_pool is None:
+                    _logger.debug("Creating unshared GrpcChannelPool.")
+                    self._grpc_channel_pool = GrpcChannelPool()
+                if self._discovery_client is None:
+                    _logger.debug("Creating unshared DiscoveryClient.")
+                    self._discovery_client = DiscoveryClient(
+                        grpc_channel_pool=self._grpc_channel_pool
+                    )
+                if self._stub is None:
+                    service_location = self._discovery_client.resolve_service(
+                        provided_interface=_V2_MEASUREMENT_SERVICE_INTERFACE,
+                        service_class=self._service_class,
+                    )
+                    channel = self._grpc_channel_pool.get_channel(service_location.insecure_address)
+                    self._stub = v2_measurement_service_pb2_grpc.MeasurementServiceStub(channel)
+        return self._stub
+
+    def _create_file_descriptor(self) -> None:
+        metadata = self._get_stub().GetMetadata(v2_measurement_service_pb2.GetMetadataRequest())
+        input_metadata = []
+        for configuration in metadata.measurement_signature.configuration_parameters:
+            input_metadata.append(
+                ParameterMetadata.initialize(
+                    display_name=configuration.name,
+                    type=configuration.type,
+                    repeated=configuration.repeated,
+                    default_value=None,
+                    annotations=dict(configuration.annotations.items()),
+                    message_type=configuration.message_type,
+                )
+            )
+        output_metadata = []
+        for output in metadata.measurement_signature.outputs:
+            output_metadata.append(
+                ParameterMetadata.initialize(
+                    display_name=output.name,
+                    type=output.type,
+                    repeated=output.repeated,
+                    default_value=None,
+                    annotations=dict(output.annotations.items()),
+                    message_type=output.message_type,
+                )
+            )
+        create_file_descriptor(
+            input_metadata=input_metadata,
+            output_metadata=output_metadata,
+            service_name=self._service_class,
+            pool=descriptor_pool.Default(),
+        )
+
+    def measure(
+        self,
+        float_in: float = 0.05999999865889549,
+        double_array_in: List[float] = [0.1, 0.2, 0.3],
+        bool_in: bool = False,
+        string_in: str = "sample string",
+        string_array_in: List[str] = ["String1", "String2"],
+        path_in: Path = r"path/test",
+        path_array_in: List[str] = ["path/test1", "path/ntest2"],
+        io_in: str = "resource",
+        pin_array_in: List[str] = ["pin1", "pin2"],
```

Comment:
Can we name this as `io_array_in` to avoid misunderstanding from the "PinArray" type which is now deprecated?